src/NewTools-DebuggerSelector/DebugRequest.class.st

```diff
@@ -0,0 +1,150 @@
+- exception: Exception
+- context: Context
+- label: String
+- process: Process
+- compiledMethod: CompiledMethod
+
+# Notes
+- You NEED to provide an exception OR a context. The other information are optional.
+- If an exception and a label are provided, the label will be ignored (in favour of the #description of the exception)
+- Providing a compiledMethod will make it so the execution will be stepped until it is in that method , prior to the debugger opening
+
+# Instance Variables
+- exception: Exception.
+- context: Context.
+- label: String. If not provided, the result of ""self defaultLabel_"" will be used.
+- process: Process. If not provided, the result of ""self defaultProcess"" will be used.
+- compiledMethod: CompiledMethod
+- preDebugActions: OrderedCollection<BlockClosure<DebugSession, nil>>. A list of blocks, taking as arguments a DebugSession on the execution to be debugged. These will be executed before the debugger opens.
+
+# Technical Details
+- the setters are meant to be used by the caller, to fill this object with information
+- the getters are meant to be used by whatever objects is opening a debugger based on the information contained in this object
+"
+Class {
+	#name : #DebugRequest,
+	#superclass : #Object,
+	#instVars : [
+		'exception',
+		'context',
+		'label',
+		'uiManager',
+		'process',
+		'compiledMethod',
+		'preDebugActions',
+		'debugSession'
+	],
+	#category : #'NewTools-DebuggerSelector-Core'
+}
+
+{ #category : #'instance creation' }
+DebugRequest class >> newForContext: aContext [
+	^ self new context: aContext; yourself
+]
+
+{ #category : #'instance creation' }
+DebugRequest class >> newForException: anException [
+	^ self new exception: anException; yourself
+]
+
+{ #category : #setter }
+DebugRequest >> compiledMethod: anObject [
+	compiledMethod := anObject
+]
+
+{ #category : #preDebugActions }
+DebugRequest >> computePreDebugActions [
+	"Fill the preDebugActions variable."
+	"If compiledMethod is not nil, this DebugRequest has been created to debug the execution of a specific method. Add a pre-debug action to step the execution until that method is reached"
+	compiledMethod ifNotNil: [ self preDebugActions add: (self preDebugActionForDebugItMethod: compiledMethod) ]
+]
+
+{ #category : #setter }
+DebugRequest >> context: anObject [
+	context := anObject
+]
+
+{ #category : #getter }
+DebugRequest >> debugSession [
+	"If a DebugSession is requested for this debug request, ask the process to create one. Subsequent calls to this method will return the same DebugSession (and not create another one)"
+	debugSession ifNil: [ debugSession := self process newDebugSessionNamed: self exception description startedAt: self exception signalerContext ].
+	^ debugSession.
+]
+
+{ #category : #'default values' }
+DebugRequest >> defaultLabel_ [
+	"The underscore is to prevent conflicting with Object>>#defaultLabel, which is a morphic method"
+	^ 'No label was provided'
+]
```

Comment:
C'est expliqué dans le commentaire de la méthode

Reply:
is still a bad and not allowed name. 
You can always `detail` your names. Also here we do not care because your DebugRequest class is not in the morphic hierarchy. You can name it `defaultLabel` without danger to have a collision (also not wanting to repeat names when the meaning is the same is incorrect)

Reply:
ok, I'll do that. Thanks